sql/core/src/main/scala/org/apache/spark/sql/execution/subquery.scala

```diff
@@ -114,9 +116,17 @@ case class InSubqueryExec(
     child: Expression,
     plan: BaseSubqueryExec,
     exprId: ExprId,
-    private var resultBroadcast: Broadcast[Array[Any]] = null) extends ExecSubqueryExpression {
-
-  @transient private var result: Array[Any] = _
+    private var resultBroadcast: Broadcast[Set[Any]] = null) extends ExecSubqueryExpression {
+
+  @transient private var result: Set[Any] = _
+  @transient private lazy val hasNull: Boolean = result.contains(null)
```

Comment:
shall we just keep an instance of `InSet` and call its `eval`?

Reply:
can reuse one `InSet` with `doGenCode`?

Reply:
yes